model/calendar.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const createError = require('http-errors');
+const debug = require('debug')('calendar:calendar');
+const Schema = mongoose.Schema;
+
+const Year = require('./year.js');
+
+const calendarSchema = Schema({
+  name: {
+    type: String,
+    required: true
+  },
+  timestamp: {
+    type: Date,
+    required: true
+  },
+  years: [
+    {
+      type: Schema.Types.ObjectId,
+      ref: 'year'
+    }
+  ]
+});
+
+const Calendar = module.exports = mongoose.model('calendar', calendarSchema);
+
+Calendar.findByIdAndAddYear = function(id, year) {
```

Comment:
If you want to add a method to your schema, you should look into using [Instance Methods](http://mongoosejs.com/docs/guide.html#methods) or [Static Methods](http://mongoosejs.com/docs/guide.html#statics).